### PR TITLE
feat: Add Bicep Contribution Repo & CI Setup Script

### PR DESCRIPTION
**Pull request by jtracey93**
# Overview/Summary

Add Bicep Contribution Repo & CI Setup Script

## This PR fixes/adds/changes/removes

1. Add Bicep Contribution Repo & CI Setup Script

### Breaking Changes

None

## As part of this Pull Request I have

- [x] Read the Contribution Guide and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Verified-Modules/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Verified-Modules/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Verified-Modules/)
- [x] Ensured PR tests are passing
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

File: docs/content/contributing/bicep/bicep-contribution-flow/_index.md [hunk 1]
```diff
@@ -61,7 +61,44 @@ When implementing the GitFlow process as described, it is advisable to configure
 
 {{< /hint >}}
 
-</br>
+## PowerShell Helper Script To Setup Fork & CI Test Environment
+
+To simplify the setup of the fork, clone and configuration of the required secrets, SPN and RBAC assignments in your Azure environment for the CI framework to function correctly in your fork we have created a PowerShell script that you can use to do steps [1](#1-setup-your-azure-test-environment), [2](#2-fork-the-module-source-repository) & [3](#3-configure-your-ci-environment) below.
```
Comment (matebarabas) [inline docs/content/contributing/bicep/bicep-contribution-flow/_index.md:66]:
```suggestion
To simplify the setup of the fork, clone and configuration of the required secrets, SPN and RBAC assignments in your Azure environment for the CI framework to function correctly in your fork, we have created a PowerShell script that you can use to do steps [1](#1-setup-your-azure-test-environment), [2](#2-fork-the-module-source-repository) & [3](#3-configure-your-ci-environment) below.
```

File: docs/content/contributing/bicep/bicep-contribution-flow/_index.md [hunk 2]
```diff
@@ -61,7 +61,44 @@ When implementing the GitFlow process as described, it is advisable to configure
...
+
+{{< hint type=important >}}
+
+You will still need to complete [step 3.3](#33-set-readwrite-workflow-permissions) manually at this time.
+
+{{< /hint >}}
+
+The script does the following steps:
```
Comment (matebarabas) [inline docs/content/contributing/bicep/bicep-contribution-flow/_index.md:74]:
```suggestion
The script performs the following steps:
```

File: docs/content/contributing/bicep/bicep-contribution-flow/_index.md [hunk 3]
```diff
@@ -61,7 +61,44 @@ When implementing the GitFlow process as described, it is advisable to configure
...
+
+1. Forks the `Azure/bicep-registry-modules` to your GitHub Account
```
Comment (matebarabas) [inline docs/content/contributing/bicep/bicep-contribution-flow/_index.md:76]:
```suggestion
1. Forks the `Azure/bicep-registry-modules` to your GitHub Account.
```

File: docs/content/contributing/bicep/bicep-contribution-flow/_index.md [hunk 4]
```diff
@@ -61,7 +61,44 @@ When implementing the GitFlow process as described, it is advisable to configure
...
+2. Clones the repo locally to your machine to the location your specify in the parameter: `-GitHubRepositoryPathForCloneOfForkedRepository`
```
Comment (matebarabas) [inline docs/content/contributing/bicep/bicep-contribution-flow/_index.md:77]:
```suggestion
2. Clones the repo locally to your machine, based on the location you specify in the parameter: `-GitHubRepositoryPathForCloneOfForkedRepository`.
```

File: docs/content/contributing/bicep/bicep-contribution-flow/_index.md [hunk 5]
```diff
@@ -61,7 +61,44 @@ When implementing the GitFlow process as described, it is advisable to configure
...
+3. Prompts you and takes you directly to the place to enable GitHub Actions Workflows on your forked repo
```
Comment (matebarabas) [inline docs/content/contributing/bicep/bicep-contribution-flow/_index.md:78]:
```suggestion
3. Prompts you and takes you directly to the place where you can enable GitHub Actions Workflows on your forked repo.
```

File: docs/content/contributing/bicep/bicep-contribution-flow/_index.md [hunk 6]
```diff
@@ -61,7 +61,44 @@ When implementing the GitFlow process as described, it is advisable to configure
...
+4. Disables all AVM module workflows, as per [Enable or Disable Workflows](/Azure-Verified-Modules/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows/)
```
Comment (matebarabas) [inline docs/content/contributing/bicep/bicep-contribution-flow/_index.md:79]:
```suggestion
4. Disables all AVM module workflows, as per [Enable or Disable Workflows](/Azure-Verified-Modules/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows/).
```

File: docs/content/contributing/bicep/bicep-contribution-flow/_index.md [hunk 7]
```diff
@@ -61,7 +61,44 @@ When implementing the GitFlow process as described, it is advisable to configure
...
+5. Creates an Azure Service Principal (SPN) and grants it RBAC roles of `User Access Administrator` & `Contributor` to the Management Group, if specified in the parameter `-GitHubSecret_ARM_MGMTGROUP_ID`, and to the Azure Subscription you provide via the parameter `-GitHubSecret_ARM_SUBSCRIPTION_ID`
+6. Creates the required GitHub Actions Secrets as per [step 3](#3-configure-your-ci-environment) in your forked repo based on the input provided in parameters and the values from resources the script creates, such as the SPN.
```
Comment (matebarabas) [inline docs/content/contributing/bicep/bicep-contribution-flow/_index.md:81]:
```suggestion
6. Creates the required GitHub Actions Secrets in your forked repo as per [step 3](#3-configure-your-ci-environment), based on the input provided in parameters and the values from resources the script creates, such as the SPN.
```

File: docs/content/contributing/bicep/bicep-contribution-flow/_index.md [hunk 8]
```diff
@@ -61,7 +61,44 @@ When implementing the GitFlow process as described, it is advisable to configure
...
+
+### Pre-requisites
+
+1. You must have the [Azure PowerShell Modules](https://learn.microsoft.com/powershell/azure/install-azure-powershell) installed and logged in with a user into the desired Tenant with permissions to create an SPN and grant RBAC over the specified Subscription and Management Group, if provided.
```
Comment (matebarabas) [inline docs/content/contributing/bicep/bicep-contribution-flow/_index.md:85]:
```suggestion
1. You must have the [Azure PowerShell Modules](https://learn.microsoft.com/powershell/azure/install-azure-powershell) installed and you need to be logged with the context set to the desired Tenant. You must have permissions to create an SPN and grant RBAC over the specified Subscription and Management Group, if provided.
```

File: docs/content/contributing/bicep/bicep-contribution-flow/_index.md [hunk 9]
```diff
@@ -61,7 +61,44 @@ When implementing the GitFlow process as described, it is advisable to configure
...
+2. You must have the [GitHub CLI](https://github.com/cli/cli#installation) installed and authenticated with the GitHub user account you wish to fork, clone and work with AVM on
```
Comment (matebarabas) [inline docs/content/contributing/bicep/bicep-contribution-flow/_index.md:86]:
```suggestion
2. You must have the [GitHub CLI](https://github.com/cli/cli#installation) installed and need to be authenticated with the GitHub user account you wish to use to fork, clone and work with on AVM.
```

File: docs/content/contributing/bicep/bicep-contribution-flow/_index.md [hunk 10]
```diff
@@ -61,7 +61,44 @@ When implementing the GitFlow process as described, it is advisable to configure
...
+
+{{< expand "New-AVMBicepBRMForkSetup.ps1 - PowerShell Helper Script" "expand/collapse" >}}
+
+The `New-AVMBicepBRMForkSetup.ps1` can be downloaded from <a href="/Azure-Verified-Modules/scripts/New-AVMBicepBRMForkSetup.ps1" download>here</a>.
+
+Once downloaded you can run the script by running - **Please change all the parameter values in the below script usage example to your own values (see the parameter documentation in the script itself)**:
```
Comment (matebarabas) [inline docs/content/contributing/bicep/bicep-contribution-flow/_index.md:92]:
```suggestion
Once downloaded, you can run the script by running the below - **Please change all the parameter values in the below script usage example to your own values (see the parameter documentation in the script itself)!**:
```

File: docs/content/contributing/bicep/bicep-contribution-flow/_index.md [hunk 11]
```diff
@@ -61,7 +61,44 @@ When implementing the GitFlow process as described, it is advisable to configure
...
+```powershell
+.\<PATH-TO-SCRIPT-DOWNLOAD-LOCATION>\New-AVMBicepBRMForkSetup.ps1 -GitHubRepositoryPathForCloneOfForkedRepository "D:\GitRepos\" -GitHubSecret_ARM_MGMTGROUP_ID "alz" -GitHubSecret_ARM_SUBSCRIPTION_ID "1b60f82b-d28e-4640-8cfa-e02d2ddb421a" -GitHubSecret_ARM_TENANT_ID "c3df6353-a410-40a1-b962-e91e45e14e4b" -GitHubSecret_TOKEN_NAMEPREFIX "ex123"
+```
+
+See the examples in the script, below the parameters, for more examples.
```
Comment (matebarabas) [inline docs/content/contributing/bicep/bicep-contribution-flow/_index.md:97]:
```suggestion
For more examples, see the below script's parameters section.
```

File: docs/content/contributing/bicep/bicep-contribution-flow/_index.md [hunk 12]
```diff
@@ -71,6 +108,12 @@ Each time in the following sections we refer to 'your xyz', it is an indicator t
 
 {{< /hint >}}
 
+{{< hint type=tip >}}
+
+Checkout the [PowerShell Helper Script](#powershell-helper-script-to-setup-fork--ci-test-environment) that can do this step automatically for you 👍
```
Comment (matebarabas) [inline docs/content/contributing/bicep/bicep-contribution-flow/_index.md:113]:
```suggestion
Checkout the [PowerShell Helper Script](#powershell-helper-script-to-setup-fork--ci-test-environment) that can do this step automatically for you! 👍
```

**Review by matebarabas — COMMENTED** [review]
@jtracey93, I reviewed the documentation part of this PR. Please ask someone else from the core team, to specifically review the script! Thanks!

**Review by rahalan — APPROVED** [review]

**Pull request merged**